tests/src/testdirichletvalue.cpp

```diff
@@ -44,20 +42,119 @@ static auto dirichletBCTest() {
   using namespace Dune::Functions::BasisFactory;
   auto basis = Ikarus::makeBasis(gridView, power<2>(lagrange<1>()));
 
-  auto basisP = std::make_shared<const decltype(basis)>(basis);
+  auto basisP    = std::make_shared<const decltype(basis)>(basis);
```

Comment:
This is some not needed legacy code. Can you please remove the shared_ptr indirection.